sklearn/linear_model/ridge.py

```diff
@@ -320,14 +320,16 @@ def ridge_regression(X, y, alpha, sample_weight=None, solver='auto',
                           "automatically changed into 'sag'.")
         solver = 'sag'
 
+    _dtype = [np.float64, np.float32]
+
     # SAG needs X and y columns to be C-contiguous and np.float64
     if solver in ['sag', 'saga']:
         X = check_array(X, accept_sparse=['csr'],
                         dtype=np.float64, order='C')
         y = check_array(y, dtype=np.float64, ensure_2d=False, order='F')
     else:
         X = check_array(X, accept_sparse=['csr', 'csc', 'coo'],
-                        dtype=np.float64)
+                        dtype=_dtype)
         y = check_array(y, dtype='numeric', ensure_2d=False)
```

Comment:
In order to force BLASS, what do you guys thing about the follwing line?

```Python
 y = check_array(y, dtype=X.dtype, ensure_2d=False)
```

we let `X` be whatever is in `_dtype` and then force X and y dtype be equal.
This is related to #8976 .